### DISCUSSION TITLE
Plasma 5 support

### DISCUSSION THREAD

**Discussion by magnouvean** [nix-community/plasma-manager#81] [Polls]
I'm unsure how we should approach supporting plasma 5 in the future. With plasma 6 just being out I'm assuming that most kde-users on nixos will be moving to plasma 6 (I have already done so), but it of course is not available for stable users yet. I am therefore wondering how we should approach plasma 5 support in the future. One easy option to try to maintain as good support as possible until plasma 6 is available in stable, and then focus only on plasma 6. However there might be people who still need to use plasma 5 in the future for some reason (hence why I am doing a poll). I would prefer to mainly focus on plasma 6 in the distant future (after plasma 6 is in stable) as there is no way I am going to be able to do appropriate testing for both 5 and 6. One thing I think we could do though is to at least create a separate branch which is compatible with plasma 5 (we don't really need to maintain it, it can just refer to commit `132e00af06dc93fa74c137533b53b083cbc1002c`, which was the last release before introducing new plasma 6 stuff). Any other ideas here?

**Comment by toast003** [comment]
I haven't updated to plasma 6 yet since I'm waiting for [my theme to update in nixpkgs](https://github.com/NixOS/nixpkgs/pull/294507) but the moment that gets merged I would switch right away

**Reply by nat-418** [reply]
I think a plasma 5 branch could easily be kept around in case any fixes etc. needed adding, no?

I would propose 5 being the default until 6 lands in stable, then 5 remains as a branch until 5 support is dropped from NixOS.

**Reply by magnouvean** [reply]
Yeah I agree. The only problem is that I'm not sure I at least have enough time to do a lot of testing for plasma 5, so it probably wont get much new high-level interfaces, but should be in a working state nonetheless. At the moment both plasma 5 and 6 should be supported, hence why no such branch has been made yet, but when we bring in the breaking changes (which probably will happen at some point) we will create a separate plasma 5 branch to keep everything working.

**Reply by nat-418** [reply]
This may be a hot take from me but this an unpaid open source project, and "support" is obviously a best-effort thing. If people use the 5 branch (as I will probably do at least for awhile longer) they can do the work to keep it updated. Otherwise, it can be there as a resource for the current userbase as-is. My two cents, at least.

**Reply by magnouvean** [reply]
Yeah I do agree there @nat-418 :)
